arches_lingo/src/arches_lingo/components/generic/ResourceInstanceRelationships.vue

```diff
@@ -0,0 +1,35 @@
+<script setup lang="ts">
+import type {
+    DataComponentMode,
+    ResourceInstanceReference,
+} from "@/arches_lingo/types";
+import ResourceInstanceRelationshipsViewer from "@/arches_lingo/components/generic/ResourceInstanceRelationshipsViewer.vue";
+import ResourceInstanceRelationshipsEditor from "@/arches_lingo/components/generic/ResourceInstanceRelationshipsEditor.vue";
+import { EDIT, VIEW } from "@/arches_lingo/constants.ts";
+
+const props = defineProps<{
+    mode?: DataComponentMode;
```

Comment:
I went back and forth on making mode optional.  By default the widgets render in view mode, since it seems like edit will be used less often.  I have no issue forcing the inclusion, though.

Reply:
Makes sense. In that case, we could leave it optional in the type and then provide a default. More explicit than baking it into the if `v-if=!props.mode || ...```

```js
const { mode = EDIT } = defineProps<{
    mode?: DataComponentMode;
```

(If your editor has inlay hints for vue turned on (I think it's default now) you'll see the magic "props." hint prepended to `mode` to clarify it came from props.)

Reply:
nice